pkg/kubelet/cm/deviceplugin/manager.go

```diff
@@ -242,8 +242,15 @@ func (m *ManagerImpl) Allocate(node *schedulercache.NodeInfo, attrs *lifecycle.P
 	pod := attrs.Pod
 	// TODO: Reuse devices between init containers and regular containers.
 	for _, container := range pod.Spec.InitContainers {
-		if err := m.allocateContainerResources(pod, &container); err != nil {
-			return err
+		for resourceKey := range container.Resources.Limits {
+			resource := string(resourceKey)
+			_, registeredResource := m.allDevices[resource]
+			_, allocatedResource := m.allocatedDevices[resource]
```

Comment:
Can this check be skipped? It was needed in `allocateContainerResources` for handling a container restart which already had resource allocated. 
Or is there a scenario where this check would be needed?

Reply:
Yep, in fact if only we add the limitation, pod with plugin resources in init-container would never succeed, so it has no chance to exist in `m.allocatedDevices`, I added it here duo to the feeling that a double check here seems more reasonable from the point of logic。
However, both are ok to me. would like to hear comments from you guys :)

Reply:
In the mentioned scenario I'd rather have an error stating that the resource is not present, signaling that the device plugin did not register yet (could be a problem worth investigating) and then once that is fixed, have the error mentioning that Device Plugin Resources are forbidden in an init container.

Reply:
@lichuqiang All right. So then, should the check be updated to identify the scenario you described and return a more appropriate error message, something like @RenaudWasTaken suggested ?

Reply:
It‘s hard to say which one of the two (prohibit requesting in init-containers and lack of registration ) is of higher priority. And if we decide to add a specific error for the registration lack scenario, we need to add it in regular containers logic too.

Reply:
@lichuqiang, IMO, `Device not registered` is bigger problem than `prohibit requesting`. First one is infra/deployment and later is a usage error.

> need to add it in regular containers ....

Fine grained error reporting is always better.